posthog/batch_exports/service.py

```diff
@@ -248,21 +251,45 @@ def create_batch_export(team_id: int, interval: str, name: str, destination_data
 
 
 async def acreate_batch_export(team_id: int, interval: str, name: str, destination_data: dict) -> BatchExport:
-    """
-    Create a BatchExport and its underlying Schedule.
-    """
+    """Create a BatchExport and its underlying Schedule."""
     return await sync_to_async(create_batch_export)(team_id, interval, name, destination_data)  # type: ignore
 
 
 def fetch_batch_export_runs(batch_export_id: UUID, limit: int = 100) -> list[BatchExportRun]:
-    """
-    Fetch the BatchExportRuns for a given BatchExport.
-    """
+    """Fetch the BatchExportRuns for a given BatchExport."""
     return list(BatchExportRun.objects.filter(batch_export_id=batch_export_id).order_by("-created_at")[:limit])
 
 
 async def afetch_batch_export_runs(batch_export_id: UUID, limit: int = 100) -> list[BatchExportRun]:
-    """
-    Fetch the BatchExportRuns for a given BatchExport.
-    """
+    """Fetch the BatchExportRuns for a given BatchExport."""
     return await sync_to_async(fetch_batch_export_runs)(batch_export_id, limit)  # type: ignore
+
+
+@async_to_sync
+async def reset_batch_export_run(temporal, batch_export_id: str | UUID, run_id: UUID) -> str:
+    """Reset an individual batch export run corresponding to a given batch export.
+
+    Resetting a workflow is considered an "advanced concept" by Temporal, hence it's not exposed
+    cleanly via the SDK, and it requries us to make a raw request.
+
+    Resetting a workflow will create a new run with the same workflow id. The new run will have a
+    reference to the original run_id that we can use to tie up re-runs with their originals.
+
+    Returns:
+        The run_id assigned to the new run.
+    """
+    request = ResetWorkflowExecutionRequest(
```

Comment:
What is the advantage of resetting over simply creating a new one via execute_workflow, given that we are resetting to the beginning?

Reply:
`execute_workflow` creates a new workflow, with a new workflow ID. So, misusing that could cause duplicate runs. Of course, we could still ensure that ourselves, but reset handles all of that for us. There isn't much of a difference between cancelling + executing vs reset, besides reset being 1 atomic operation.